src/MediatR/Mediator.cs

```diff
@@ -21,17 +21,32 @@ public class Mediator : IMediator
         private readonly ConcurrentDictionary<Type, Type> _genericHandlerCache;
         private readonly ConcurrentDictionary<Type, Type> _wrapperHandlerCache;
 
+        private PublishAsyncOptions _publishOption;
+
         /// <summary>
         /// Initializes a new instance of the <see cref="Mediator"/> class.
         /// </summary>
         /// <param name="singleInstanceFactory">The single instance factory.</param>
         /// <param name="multiInstanceFactory">The multi instance factory.</param>
-        public Mediator(SingleInstanceFactory singleInstanceFactory, MultiInstanceFactory multiInstanceFactory)
+        public Mediator(SingleInstanceFactory singleInstanceFactory, MultiInstanceFactory multiInstanceFactory) :

```

Comment:
I hadn't thought about the impact of adding an extra constructor on DI resolution. 

Like I said in my comment on overloads of PublishAsync, I'd like to make this publish behaviour a global option so that I don't have to update every call site where events are published.

I've just checked out the [breaking change rules for CoreFx](https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/breaking-change-rules.md) and by my reading adding an extra constructor wouldn't be considered a breaking change.

However, if the extra constructor issue is serious, perhaps we could have a property on Mediator to configure the default behaviour?